noc/uninitialized_value_concatenation

denissafiullin-deriv opened the pull request:
Currently `$log->debug` produces a warning due to the uninitialised value of `$c->tx->remote_address`. During the last month, there were 277 warn logs out of 1.04 million connections, which is 0.027 %. To solve the issue, the following steps were taken:
1. Use `Log::Any` instead of `Mojo::Log` to be able to use `debugf` instead of `debug` and to follow the [Perl Style Guide](https://wikijs.deriv.cloud/en/IT-Department/style-guide/perl/guide#:~:text=Copy-,%C2%B6,-Logging).
2. Avoid string interpolation as discussed [here](https://deriv-group.slack.com/archives/C0D442KHV/p1701738911448829)
3. Use single quotes instead of double quotes, as interpolation is not required
4. Fallback to string `undef` if `remote_address` wasn't resolved.

ClickUp card: https://app.clickup.com/t/20696747/PLA-768

denissafiullin-deriv committed:
Add fallback value for remote_address and make tidy

denissafiullin-deriv committed:
noc/uninitialized_value_concatenation

denissafiullin-deriv committed:
noc/uninitialized_value_concatenation

review-deriv reviewed (commented):
PR description is too short. You can use this format:

```
# Description
This is a summary of the changes. This should describe why do we need this change.
Whys and Hows come here.

# Cards
* https://clickup.com/12345
* https://clickup.com/12346

```

review-deriv reviewed (commented):
PR description is too short. You can use this format:

```
# Description
This is a summary of the changes. This should describe why do we need this change.
Whys and Hows come here.

# Cards
* https://clickup.com/12345
* https://clickup.com/12346

```

denissafiullin-deriv committed:
trigger tests 2023-11-29 06:47:51

mariovw-deriv commented on lib/Mojo/WebSocketProxy/Dispatcher.pm:
```diff
@@ -51,7 +51,7 @@ sub open_connection {
 
     my $log = $c->app->log;
-    $log->debug("accepting a websocket connection from " . $c->tx->remote_address);
+    $log->debug("accepting a websocket connection from " . ($c->tx->remote_address // 'undefined remote address'));
```
can we use debugf here?

denissafiullin-deriv replied:
Yes, replaced with `debugf`

denissafiullin-deriv committed:
replace debug with debugf

denissafiullin-deriv committed:
correct typo

denissafiullin-deriv committed:
trigger tests 2023-12-06 06:46:52

aisha-deriv commented on t/29-request_logger_test.t:
```diff
@@ -12,7 +12,7 @@ subtest 'Test log_message method' => sub {
+    $logger->errorf('This is an error %s message', "with params");
     $logger->debugf('This is debug message with %s', 'params');
     pass('All log levels tested');
 };
```
why are the changes in requestlogger and test file are you running tidy for these statements ?

denissafiullin-deriv replied:
I ran `tidy` for the whole repo. This correction looks valid to me.

denissafiullin-deriv committed:
return to debug from debugf, as debugf is not supported by Mojo::Log, which is used in all tests

denissafiullin-deriv committed:
return to debug from debugf, as debugf is not supported by Mojo::Log, which is used in all tests

aisha-deriv commented on lib/Mojo/WebSocketProxy/Dispatcher.pm:
```diff
@@ -52,7 +52,7 @@ sub open_connection {
     my $remote_address = $c->tx->remote_address // 'undef';
-    $log->debugf('accepting a websocket connection from ' . $remote_address);
+    $log->debug('accepting a websocket connection from ' . $remote_address);
     # Enable permessage-deflate
```
interpolation is not recommended https://deriv-group.slack.com/archives/C0D442KHV/p1701738911448829
can we show log in case its defined only

denissafiullin-deriv replied:
Thank you. Took out string interpolation

chylli-deriv commented on lib/Mojo/WebSocketProxy/Dispatcher.pm:
```diff
@@ -50,8 +50,9 @@ sub ok {
-    $log->debug("accepting a websocket connection from " . $c->tx->remote_address);
+    my $log            = $c->app->log;
+    my $remote_address = $c->tx->remote_address // 'undef';
+    $log->debug('accepting a websocket connection from ' . $remote_address);
```
log::any format log can handle undef
```suggestion
    $log->debugf('accepting a websocket connection from %s', $c->tx->remote_address);
```

denissafiullin-deriv replied:
Thank you. Introduced `Log::Any`

denissafiullin-deriv committed:
implement Log::Any and avoid string interpolation

denissafiullin-deriv committed:
Add changelog in Changes file

denissafiullin-deriv committed:
More descriptive wording in Change file

aisha-deriv committed:
v0.16

    Mojo::WebSocketProxy::Dispatcher changes:
    - Use Log::Any instead of Mojo::Log in Dispatcher;
    - Avoid string interpolation to solve warning due to the uninitialised value of $c->tx->remote_address;
    - Add fallback value for cases when remote_address can't be resolved;

aisha-deriv committed:
trigger tests 2023-12-15 10:29:19

aisha-deriv committed:
trigger tests 2023-12-15 10:57:11
